src/frontend/src/optimizer/plan_node/batch_sys_seq_scan.rs

```diff
@@ -0,0 +1,224 @@
+        let order = if scan_ranges.len() > 1 {
+            Order::any()
+        } else {
+            core.get_out_column_index_order()
+        };
+        let base = PlanBase::new_batch_with_core(&core, dist, order);
+
+        {
+            // validate scan_range
+            scan_ranges.iter().for_each(|scan_range| {
+                assert!(!scan_range.is_full_table_scan());
+                let scan_pk_prefix_len = scan_range.eq_conds.len();
+                let order_len = core.table_desc.order_column_indices().len();
+                assert!(
+                    scan_pk_prefix_len < order_len
+                        || (scan_pk_prefix_len == order_len && is_full_range(&scan_range.range)),
+                    "invalid scan_range",
+                );
+            })
+        }
+
+        Self {
+            base,
+            core,
+            scan_ranges,
+        }
+    }
+
+    pub fn new(core: generic::SysScan, scan_ranges: Vec<ScanRange>) -> Self {
+        // Use `Single` by default, will be updated later with `clone_with_dist`.
+        Self::new_inner(core, Distribution::Single, scan_ranges)
+    }
+
+    fn clone_with_dist(&self) -> Self {
+        Self::new_inner(
+            self.core.clone(),
+            Distribution::Single,
+            self.scan_ranges.clone(),
+        )
+    }
+
+    /// Get a reference to the batch seq scan's logical.
+    #[must_use]
+    pub fn core(&self) -> &generic::SysScan {
+        &self.core
+    }
+
+    pub fn scan_ranges(&self) -> &[ScanRange] {
+        &self.scan_ranges
+    }
+
+    fn scan_ranges_as_strs(&self, verbose: bool) -> Vec<String> {
+        let order_names = match verbose {
+            true => self.core.order_names_with_table_prefix(),
+            false => self.core.order_names(),
+        };
+        let mut range_strs = vec![];
+
+        let explain_max_range = 20;
+        for scan_range in self.scan_ranges.iter().take(explain_max_range) {
+            #[expect(clippy::disallowed_methods)]
+            let mut range_str = scan_range
+                .eq_conds
+                .iter()
+                .zip(order_names.iter())
+                .map(|(v, name)| match v {
+                    Some(v) => format!("{} = {:?}", name, v),
+                    None => format!("{} IS NULL", name),
+                })
+                .collect_vec();
+            if !is_full_range(&scan_range.range) {
+                let i = scan_range.eq_conds.len();
+                range_str.push(range_to_string(&order_names[i], &scan_range.range))
+            }
+            range_strs.push(range_str.join(" AND "));
+        }
+        if self.scan_ranges.len() > explain_max_range {
+            range_strs.push("...".to_string());
+        }
+        range_strs
+    }
+}
+
+impl_plan_tree_node_for_leaf! { BatchSysSeqScan }
+
+fn lb_to_string(name: &str, lb: &Bound<ScalarImpl>) -> String {
```

Comment:
These auxiliary functions should be reused.

Reply:
Agree, there are several functions which can be reused. Intentionally leave it out for future PR to reduce conflicts. Tracked: https://github.com/risingwavelabs/risingwave/issues/13146#issuecomment-1805229754